src/main/java/com/GeorgesServer/app/EstablishesConnection.java

```diff
@@ -3,25 +3,29 @@
 import java.io.*;
 import java.net.ServerSocket;
 import java.net.Socket;
-import java.nio.Buffer;
 
 public class EstablishesConnection {
 
-    private SocketFactory socketFactory;
+    private ServerSocket serverSocket;
 
-    public EstablishesConnection (SocketFactory socketFactory) {
-        this.socketFactory = socketFactory;
+    public EstablishesConnection (ServerSocket serverSocket) {
+
+        this.serverSocket = serverSocket;
     }
 
-    public Connections connect(int port) {
-        ServerSocket serverSocket = socketFactory.createSocket(port);
-        Socket clientSocket = socketFactory.createClientSocket(serverSocket);
-        InputStream inputStream = socketFactory.createInputStream(clientSocket);
-        OutputStream outputStream = socketFactory.createOutPutStream(clientSocket);
-        OutputStreamWriter outputStreamWriter = socketFactory.createOutputStreamWriter(outputStream);
-        InputStreamReader inputStreamReader = socketFactory.createInputStreamReader(inputStream);
-        BufferedReader bufferedReader = socketFactory.createBufferedReader(inputStreamReader);
+    public Connections connect() {
 
-        return new Connections(bufferedReader, outputStreamWriter);
+        try {
+            Socket clientSocket = this.serverSocket.accept();
+            InputStream inputStream = clientSocket.getInputStream();
+            OutputStream outputStream = clientSocket.getOutputStream();
+            InputStreamReader inReader = new InputStreamReader(inputStream);
+            OutputStreamWriter outWriter = new OutputStreamWriter(outputStream);
+            BufferedReader buffReader = new BufferedReader(inReader);
+            return new Connections(buffReader, outWriter);
+        } catch (IOException e) {
+            e.printStackTrace();
+            return null;
```

Comment:
There are a few of these in here. Not the end of the world just now, but when you notice null checks spreading through your code it's time for an abstraction